conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DefaultStaticFactoryMethodGenerator.java

```diff
@@ -127,6 +142,45 @@ public MethodSpec generate(ServiceDefinition def) {
         return method;
     }
 
+    private MethodSpec createHelperToConstructExceptionDeserializerArgs(String serviceDefinitionPackageName) {
+        List<ErrorDefinition> errorDefinitions = packageToErrorDefinitionsMapping.get(serviceDefinitionPackageName);
+        TypeVariableName typeVariableT = TypeVariableName.get("T");
+        MethodSpec.Builder methodBuilder = MethodSpec.methodBuilder("createExceptionDeserializerArgs")
+                .addTypeVariable(typeVariableT)
+                .addModifiers(Modifier.PRIVATE)
+                .returns(ParameterizedTypeName.get(ClassName.get(ExceptionDeserializerArgs.class), typeVariableT))
+                .addParameter(ParameterizedTypeName.get(ClassName.get(TypeMarker.class), typeVariableT), "returnType");
+
+        CodeBlock.Builder exceptions = CodeBlock.builder()
+                .add("return $T.<$T>builder()", ExceptionDeserializerArgs.class, typeVariableT)
+                .add(".returnType(returnType)");
+        // Add exceptions from error definitions
+        for (ErrorDefinition errorDef : errorDefinitions) {
+            String errorName = errorDef.getErrorName().getName();
+            ClassName errorClass = ClassName.get(
+                    Packages.getPrefixedPackage(errorDef.getErrorName().getPackage(), options.packagePrefix()),
+                    ErrorGenerationUtils.errorTypesClassName(errorDef.getNamespace()),
```

Comment:
style nit: these are reused, so let's extract (might also consider moving the various `ClassName` construction pieces to helper methods within `ErrorDefinition`?)